edflow/custom_logging.py

```diff
@@ -425,15 +418,17 @@ def _create_logger(name, out_dir, level=logging.INFO):
         logger = logging.getLogger(name)
         logger.setLevel(level)
 
-        def _colored_log(level, msg, *args, color=None, **kwargs):
-            extra = kwargs.get("extra", {})
-            extra["color"] = color
+        if not hasattr(logger, "log_orig"):
```

Comment:
Nice! If I understand it correctly, you are implementing a fallback here, that makes sure the unaltered log can also be generated, right? Is log_orig called at some point? E.g. when writing to the log file?

Reply:
yes it's called by _colored_log which overwrites the _log method which is called by the actual logger methods like info etc. Prior to this PR, a first instantiation set _log_orig to _log (the original log) but a second instantiation then set _log_orig again to _log which was then already overwritten by _colored_log which then tried to call itself resulting in an infinite recursion